src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -184,6 +181,8 @@ class DeepDiveLogParser extends JavaTokenParsers {
     | "(" ~> expr <~ ")"
     )
 
+  def pattern : Parser[Expr] = "_" ^^ { case _ => Placeholder() } | expr
```

Comment:
Now, `pattern` should be of type `Parser[Patt]` instead with the three alternative cases.